Makefile

```diff
@@ -1,5 +1,7 @@
-main_solver: ./src/solver/main_solver.cpp ./src/components/token.cpp ./src/components/order.cpp ./src/api/aux.cpp
-	g++ -Wall ./src/solver/main_solver.cpp ./src/components/token.cpp ./src/components/order.cpp ./src/components/amm.cpp ./src/api/aux.cpp ./src/solver/optimize.cpp -o ./src/solver/main_solver -lgmp
+solver: ./src/solver/main_solver.cpp ./src/components/token.cpp ./src/components/order.cpp ./src/api/aux.cpp
+	g++ -Wall ./src/solver/main_solver.cpp ./src/components/token.cpp ./src/components/order.cpp ./src/components/amm.cpp ./src/api/aux.cpp ./src/solver/optimize.cpp -o ./src/solver/solver -lgmp
+
+solver_server: ./src/solver/main_solver_server.cpp ./src/components/token.cpp ./src/components/order.cpp ./src/api/aux.cpp
+	g++ -Wall -I ~/spdlog/spdlog/include/ -I ./external_libs/ ./src/solver/main_solver_server.cpp ./src/components/token.cpp ./src/components/order.cpp ./src/components/amm.cpp ./src/api/aux.cpp ./src/solver/optimize.cpp -o ./src/solver/solver_server -lgmp -lpthread
+
 
```

Comment:
Just noting that the CMake build system is the way to go these days. I agree that for a simple project this is enough.

Reply:
Thanks for the suggestion. I am planning to eventually replace this toy makefile with CMake, but was hoping to first have a first working version, and then do some minor updates such as replacing this makefile at the very end.